users/kimurayoshimi/10_2.js

```diff
@@ -1,13 +1,15 @@
 const a = (b) =>
-  b.reduce((target, key, index) => {
-    target.ids = [...([target.ids] + key.id)];
-    console.log(target.ids[index]);
-    target.entities = {
-      ...target.entities,
-      [key.id]: { value: key.value },
-    };
-    return target;
-  }, {});
+  b.reduce(
+    (target, key, index) => {
+      target.ids[index] = key.id;
+      target.entities = {
+        ...target.entities,
+        [key.id]: { value: key.value },
+      };
+      return target;
+    },
+    { ids: [] }
+  );
```

Comment:
I changed the code to pass an empty array as the initial value and put the key.id in the [index] of the array.
```js
   target.ids[index] = key.id;
```